libs/traits/src/lib.rs

```diff
@@ -687,3 +687,44 @@ pub trait CurrencyInspect {
 	/// Checks whether the provided currency is a tranche token.
 	fn is_tranche_token(currency: Self::CurrencyId) -> bool;
 }
+
+pub trait TokenSwaps<Account> {
+	type CurrencyId;
+	type Balance;
+	type OrderId;
+	/// Swap tokens buying a `buy_amount` of `currency_in` using the
+	/// `currency_out` tokens. The implementator of this method should know
+	/// the current market rate between those two currencies.
+	/// `sell_price_limit` defines the lowest price acceptable for
```

Comment:
I understand `sell_price_limit` as a way of protecting my order against a big conversion ratio. 

So, if I want to buy 10 selling 20, it will be ok if `sell_price_limit` is 30, but if the conversion drops to 0.25, then I would not sell 40 to buy 10. (double-check this @thea-leake)

> Would this be the minimum amount of currency_in a user wants to get for 1 unit of currency_out?

I understand define it as the maximum amount of `currency_out` I would give for getting the `buy_amount` of `currency_in`

Reply:
as we use it as `buy_amount.ensure_mul(sell_price_limit)` I think it is not as you described it above, right? In your example it would rather be
* `buy_amount = 10.00`
* `sell_price_limit = 0.5`

But I think we are having a logical bug here.